test/apps/Makefile.am

```diff
@@ -122,9 +125,8 @@ endif
 .cu.o : ; $(NVCC) -c -o $@ $<
 
 test_cuda_get_symbol_address.o: test_cuda_get_symbol_address.cu
-	$(NVCC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(test_cuda_get_symbol_address_CPPFLAGS) -MT $@ -MD -MF \
-	$(DEPDIR)/test_cuda_get_symbol_address.Tpo -c -o $@ \
-	`test -f '$<' || echo '$(srcdir)/'`$<
-	$(AM_V_at)$(am__mv) $(DEPDIR)/test_cuda_get_symbol_address.Tpo \
-	$(DEPDIR)/test_cuda_get_symbol_address.Po
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(test_cuda_get_symbol_address_COMPILE) -MT $@ -MD -MF $(test_cuda_get_symbol_address_DEPBASE).Tpo -c -o $@ `test -f '$(<F)' || echo '$(srcdir)/'`$(<F)
+@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(test_cuda_get_symbol_address_DEPBASE).Tpo $(test_cuda_get_symbol_address_DEPBASE).Po
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='$(<F)' object='$@' libtool=no @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(test_cuda_get_symbol_address_COMPILE) -Xcompiler "-fPIC" -c -o $@ `test -f '$(<F)' || echo '$(srcdir)/'`$(<F)
```

Comment:
can we simplify this further? for example, do we readlly need  am__fastdepCC_TRUE/am__fastdepCC_FALSE ? seems it switches between 2 modes of deps, can we use just one?

Reply:
Yes, these variables are replaced by `#` to comment out lines depending on the configure option (dependency-tracking) and the compiler version.
We need these variables at the beginning of the line to compile the object file in the same way as all other object files.